californium-core/src/main/java/org/eclipse/californium/core/CoapClient.java

```diff
@@ -322,6 +315,19 @@ public CoapClient setExecutor(ExecutorService executor, boolean detach) {
 		return this;
 	}
 
+	private ScheduledThreadPoolExecutor getSecondaryExecutor() {
+		if (secondaryExecutor == null) {
+			synchronized (this) {
+				if (secondaryExecutor == null) {
+					secondaryExecutor = new ScheduledThreadPoolExecutor(1,
```

Comment:
1. any reason not to use the `ExecutorsUtil.newDefaultSecondaryScheduler`?
2. the "double check" pays off?

Reply:
1. Because for a client I feel that 1 thread was enough by default.
2. I did that to avoid to go in the synchronized block each time we call `observe`. Do you have any idea to avoid it ? maybe you propose to always  go in the synchronized block ?

Reply:
2. I guess, running the benchmarks will show no difference, if the double check is removed.

Reply:
Currently I tried to run benchmarks on 2.0.x branch (so without my modifications) and it blocks on coap+tcp run :/

Do you want we test it before to remove it or do you want I remove it now ? (personally I have no strong opinion about that)

Reply:
With netty 4.1.34? If so, please update 2.0.x again. If not, just comment out the tcp stuff in the benchmark.sh.